superset-frontend/src/explore/components/Control.jsx

```diff
@@ -70,16 +75,18 @@ export default class Control extends React.PureComponent {
     this.setState({ hovered });
   }
   render() {
-    if (!this.props.type) return null; // this catches things like <hr/> elements (not a control!) shoved into the control panel configs.
-    const ControlType = controlMap[this.props.type];
+    const { type: controlType } = this.props;
+    if (!controlType) return null; // this catches things like <hr/> elements (not a control!) shoved into the control panel configs.
```

Comment:
Side note, those really rub me the wrong way. At some point, I hope to open a PR that provides an alternate solution to that.

Reply:
I think this comment is inaccurate. The JSX elements and nulls should be handled by [ControlPanelContainer.jsx](https://github.com/ktmud/incubator-superset/blob/efd0314819b91ebc00af1cbd4e32cb0b997d40e1/superset-frontend/src/explore/components/ControlPanelsContainer.jsx#L132-L137) already.